wsdl/ver10/schema/onvif.xsd

```diff
@@ -7631,6 +7631,131 @@ and sample rate. </xs:documentation>
 		<xs:anyAttribute processContents="lax"/>
 	</xs:complexType>
 	<!--===============================-->
+	<xs:simpleType name="TargetFormat">
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="MP4">
+				<xs:annotation>
+					<xs:documentation>
+						MP4 files with all tracks in a single file.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+			<xs:enumeration value="CMAF">
+				<xs:annotation>
+					<xs:documentation>
+						CMAF compliant MP4 files with 1 track per file.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+		</xs:restriction>
+	</xs:simpleType>
+	<!--===============================-->
+	<xs:simpleType name="EncryptionMode">
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="CENC">
+				<xs:annotation>
+					<xs:documentation>
+						AES-CTR mode full sample and video NAL Subsample encryption, defined in ISO/IEC 23001-7.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+			<xs:enumeration value="CBCS">
+				<xs:annotation>
+					<xs:documentation>
+						AES-CBC mode partial video NAL pattern encryption, defined in ISO/IEC 23001-7.
+					</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+		</xs:restriction>
+	</xs:simpleType>
+	<!--===============================-->
+	<xs:complexType name="RecordingEncryption">
+		<xs:sequence>
+			<xs:element name="KID" type="xs:hexBinary">
```

Comment:
Hi Felix, in advanced security, KeyID are usually `tas:KeyID` types, but here they are hexBinary everywhere. Is this expected?

Reply:
ISO/IEC 23001-7 defines key IDs to be 16 bytes (and recommends to use UUIDs as key IDs). `tas:KeyID` is defined as `xs:NCName` with a length of up to 64 characters. I think it makes sense to store the key ID in its hexadecimal representation. But maybe we should also specify the length here?

Reply:
No, I think its fine. I just didn't understand the link with UUIDs. Maybe we should have an informative field in the WSDL indicating the suggestion?

Reply:
Hi JF, maybe we should define a different element for specifying the key ID of a certificate in the future? Maybe we can discuss this in tomorrow's call.

Reply:
As discussed during our call, I changed the type of `KID` to `xs:string`. This way, it can hold a hexadecimal key ID for the current encryption modes and something different (like a `tas:KeyID`) in the future.

Reply:
Hi Felix,

I was discussing encryption with Maxime today and we realized that we should enforce the KID to be a UUID through the xs:hexBinary type (minLength=32 & maxLength=32) so that devices can use the KID in the ProtectionSystemSpecificHeaderBox (pssh). See section 8.1.2 from ISO/IEC 23001-7)